src/helpers/parsing/arguments/index.ts

```diff
@@ -0,0 +1,54 @@
+import {
+  DirectiveNode,
+  FieldNode,
+  ListValueNode,
+  ObjectValueNode,
+  ValueNode,
+} from "graphql";
+
+import { ScalarValueNode } from "../types";
+import { ObjectMap } from "../../../types";
+
+type ParseValueResult = string | boolean | null | ObjectMap | any[];
+
+function parseValue(valueNode: ValueNode): ParseValueResult {
+  let output: ParseValueResult;
+
+  if (valueNode.hasOwnProperty("value")) {
+    const scalarValueNode = valueNode as ScalarValueNode;
+    output = scalarValueNode.value;
+  } else if (valueNode.kind === "ObjectValue") {
+    const objectValueNode = valueNode as ObjectValueNode;
+    const obj: ObjectMap = {};
+
+    objectValueNode.fields.forEach(({ name, value }) => {
+      obj[name.value] = parseValue(value);
+    });
+
+    output = obj;
+  } else if (valueNode.kind === "ListValue") {
+    const listValueNode = valueNode as ListValueNode;

```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this unnecessary cast. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS4325)